lib/experimental/fonts/font-library/class-wp-rest-font-faces-controller.php

```diff
@@ -228,7 +228,99 @@ public function get_item( $request ) {
 			);
 		}
 
-		return $font_face;
+		return $response;
+	}
+
+	/**
+	 * Creates a font face for the parent font family.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_REST_Response|WP_Error Response object on success, or WP_Error object on failure.
+	 */
+	public function create_item( $request ) {
+		// Create the font face post so we have an ID to attach the font files to.
+		$font_face_id = wp_insert_post(
+			array(
+				'post_type'   => $this->post_type,
+				'post_parent' => $request['parent'],
+				'post_status' => 'publish',
+				'post_title'  => 'Font Face',
+				'post_name'   => 'wp-font-face',
+			)
+		);
+
+		$font_face_settings = json_decode( $request->get_param( 'font_face_settings' ), true );
+		$file_params        = $request->get_file_params();
+
+		// Move the uploaded font asset from the temp folder to the fonts directory.
+		if ( ! function_exists( 'wp_handle_upload' ) ) {
+			require_once ABSPATH . 'wp-admin/includes/file.php';
+		}
+
+		if ( is_string( $font_face_settings['src'] ) ) {
+			$file      = $file_params[ $font_face_settings['src'] ];
+			$font_file = $this->handle_font_file_upload( $file );
+			if ( isset( $font_file['error'] ) ) {
+				return new WP_Error(
+					'rest_font_upload_unknown_error',
+					$font_file['error'],
+					array( 'status' => 500 )
+				);
+			}
+			$font_face_settings['src'] = $font_file['url'];
+			$font_relative_path        = _wp_relative_fonts_path( $font_file['file'] );
+			add_post_meta( $font_face_id, '_wp_font_face_file', $font_relative_path );
+		} else {
+			foreach ( $font_face_settings['src'] as $index => $src ) {
+				$file      = $file_params[ $src ];
+				$font_file = $this->handle_font_file_upload( $file );
+				if ( isset( $font_file['error'] ) ) {
+					return new WP_Error(
+						'rest_font_upload_unknown_error',
+						$font_file['error'],
+						array( 'status' => 500 )
+					);
+				}
+				$font_face_settings['src'][ $index ] = $font_file['url'];
+
+				$font_relative_path = _wp_relative_fonts_path( $font_file['file'] );
+				add_post_meta( $font_face_id, '_wp_font_face_file', $font_relative_path );
+			}
+		}
+
+		wp_update_post(
+			array(
+				'ID'           => $font_face_id,
+				'post_content' => wp_json_encode( $font_face_settings ),
+			)
+		);
+
+		$font_face_post = get_post( $font_face_id );
+
+		return $this->prepare_item_for_response( $font_face_post, $request );
+	}
+
+	protected function handle_font_file_upload( $file ) {
+		add_filter( 'upload_mimes', array( 'WP_Font_Library', 'set_allowed_mime_types' ) );
+		add_filter( 'upload_dir', array( 'WP_Font_Library', 'set_upload_dir' ) );
+
+		$overrides = array(
+			// Not testing a form submission.
+			'test_form' => false,
+			// Seems mime type for files that are not images cannot be tested.
+			// See wp_check_filetype_and_ext().
+			'test_type' => false,
```

Comment:
The mime times passed via this overrides object don't seem to matter, however the `set_allowed_mime_types` filter seems to achieve what we need allowing the mime types to be tested.

I believe the above value should be true, however I think the 'mimes' value can be eliminated.

Reply:
Thanks! It does seem to work without the `mimes` override here, so I've removed it.